core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -920,7 +921,7 @@ object IO extends Serializable {
    * val nanoTime: IO[Nothing, Long] = IO.sync(System.nanoTime())
    * }}}
    */
-  final def syncSubmit[A](effect: ExecutionContext => A): IO[Nothing, A] = new SyncEffect[A](effect)
+  final def syncSubmit[A](effect: Env => A): IO[Nothing, A] = new SyncEffect[A](effect)
```

Comment:
This should probably now be `submit0`, and have variants `syncSubmit: (EC => A) => IO[Nothing, A]` and `asyncSubmit` that use `env.executor(Asynchronous).asExecutionContext`

Reply:
Also this is pretty powerful! Previously we couldn't wrap Java libraries that require impure callbacks without carrying the RTS around. This will let us use `unsafeRunSync` inside impure callbacks. 👍

Reply:
The question is what execution context to give them, and why? If you give them `Asynchronous` and then end up doing blocking IO, then you're in trouble... It could be dictated "top down", i.e. give them the execution context of the parent, whether that's sync or async. Or fixed in advance to something.

Needs some thought.